src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -207,9 +210,12 @@ class MarathonSchedulerActor private (
     case msg => logger.warn(s"Received unexpected message from ${sender()}: $msg")
   }
 
-  def scaleRunSpecs(): Unit = {
-    groupRepository.root().foreach { root =>
-      root.transitiveRunSpecs.foreach(spec => self ! ScaleRunSpec(spec.id))
+  def scaleRunSpecs(): Future[Done] = {
+    implicit val timeout: Timeout = Timeout(10.seconds)
```

Comment:
We probably do not want to hardcode this timeout here. What about the 30 seconds timeout?